src/clr-angular/forms/datepicker/date-input.ts

```diff
@@ -0,0 +1,195 @@
+                // This makes sure that ngModelChange is fired
+                // TODO: Check if there is a better way to do this.
+                // NOTE: Its important to use NgControl and not NgModel because
+                // NgModel only works with template driven forms
+                if (this._ngControl) {
+                    this._ngControl.control.setValue(dateStr);
+                }
+            }));
+            this._subscriptions.push(this._dateIOService.dateUpdated.subscribe(() => {
+                this._dateUpdated.emit(this._dateIOService.date);
+            }));
+        }
+    }
+
+    /**
+     * Writes the date string value to the input field
+     */
+    private writeDateStrToInputField(value: string): void {
+        this.renderer.setProperty(this.elRef.nativeElement, "value", value);
+    }
+
+    /**
+     * Sets the Input Date on the DateIOService
+     */
+    set inputDate(value: string) {
+        this._dateIOService.inputDate = value;
+    }
+
+    /**
+     * Processes a date object to check if its valid or not.
+     */
+    private processDate(value: Date): void {
+        if (this._dateIOService && value) {
+            // The date object is converted back to string because in Javascript you can create a date object
+            // like this: new Date("Test"). This is a date object but it is invalid. Converting the date object
+            // that the user passed helps us to verify the validity of the date object.
+            this._dateIOService.inputDate = this._dateIOService.toLocaleDisplayFormatString(value);
+            if (this._dateIOService.inputDate && !this.initialLoad) {
+                this.writeDateStrToInputField(this._dateIOService.inputDate);
+            }
+        }
+    }
+
+    private initialLoad: boolean = true;
+    private dateValueOnInitialLoad: Date;
+
+    /**
+     * Javascript Date object input set by the user.
+     */
+    @Input("clrDate")
+    set date(value: Date) {
+        if (this.initialLoad) {
+            // Store date value passed by the user to process after the services have been initialized by
+            // the ngOnInit hook.
+            this.dateValueOnInitialLoad = value;
+        } else {
+            this.processDate(value);
+        }
+    }
+
+    @Output("clrDateChange") _dateUpdated: EventEmitter<Date> = new EventEmitter<Date>(false);
+
+    /**
+     * Returns the date format for the placeholder according to which the input should be entered by the user.
+     */
+    @HostBinding("attr.placeholder")
+    get placeholderText(): string {
+        return this._dateIOService.placeholderText;
+    }
+
+    /**
+     * Sets the input type to text when the datepicker is enabled. Reverts back to the native date input
+     * when the datepicker is disabled. Datepicker is disabled on mobiles.
+     */
+    @HostBinding("attr.type")
+    get inputType(): string {
+        return this._datepickerEnabledService.isEnabled ? "text" : "date";
```

Comment:
Let's do a platform check on this binding to only activate on browser, otherwise leave it as it was set.

Reply:
done